### PR TITLE
Add Autoprefixer postcss Plugin

### PR DESCRIPTION
**Pull request by coliff**
Fixes: #4157

Confirmed this is working as expected. The generated CSS in `dist` for user-select includes this now:

```css
  -webkit-user-select: none;
  -moz-user-select: none;
  user-select: none;
```

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -185,7 +186,8 @@
     "gl-stats": "node --no-warnings --loader ts-node/esm test/bench/gl-stats.ts",
     "prepare": "npm run codegen",
     "typecheck": "tsc --noEmit && tsc --project tsconfig.dist.json",
-    "tsnode": "node --experimental-loader=ts-node/esm --no-warnings"
+    "tsnode": "node --experimental-loader=ts-node/esm --no-warnings",
+    "spellcheck": "npx cspell \"**/*.*s\" --no-progress"
```
Comment (HarelM) [inline package.json:190]: Doesn't this spellcheck requires a config file? 
I would advise to move this to a separate PR.

File: src/css/maplibre-gl.css [hunk 2]
```diff
@@ -23,9 +23,6 @@
 .maplibregl-canvas-container.maplibregl-interactive,
 .maplibregl-ctrl-group button.maplibregl-ctrl-compass {
     cursor: grab;
-    -moz-user-select: none;
```
Comment (HarelM) [inline src/css/maplibre-gl.css:26]: Are there other places in the file that we can clean as part of this PR?

File: package.json (hunk 1 above)
Comment (coliff) [reply package.json:190]: yep, sorry, left that in error. I'll remove...

File: src/css/maplibre-gl.css (hunk 2 above)
Comment (coliff) [reply src/css/maplibre-gl.css:26]: nope, that's all of it I think.

**Review by HarelM — APPROVED** [review]
